packages/@aws-cdk/aws-stepfunctions-tasks/README.md

```diff
@@ -211,6 +213,25 @@ runtime to use to evaluate the expression. Currently, the only runtime
 supported is `lambda.Runtime.NODEJS_10_X`.
 
 
+## API Gateway
+
+Step Functions supports [API Gateway](https://docs.aws.amazon.com/step-functions/latest/dg/connect-api-gateway.html) through the service integration pattern.
```

Comment:
just confirming: this is not the same as `apigatewayv2` correct?

Reply:
Not the same.

Reply:
But this works to invoke functions defined via apigatewayv2, correct?

Reply:
haven't played with it myself yet, can you clarify @Sumeet-Badyal

Reply:
From this page - https://docs.aws.amazon.com/step-functions/latest/dg/connect-api-gateway.html

> You use Amazon API Gateway to create, publish, maintain, and monitor HTTP and REST APIs. To integrate with API Gateway, you define a Task state in Step Functions that directly calls an API Gateway HTTP or API Gateway REST endpoint, without writing code or relying on other infrastructure.

This task supports invoking Rest APIs (v1) and HTTP APIs (v2) but not websocket APIs (v2).

Given our modeling of higher level constructs, we'll need to implement this as two tasks (not necessarily in the same PR). One that takes `IRestApi` from the `@aws-cdk/aws-apigateway` module and another that takes `IHttpApi` from the `@aws-cdk/aws-apigatewayv2` module.

Reply:
Ah; looks like someone still needs to add IHttpApi too

Reply:
@Sumeet-Badyal are you still working on this? Any plans to add the second task for the IHttpApi case? Else when my aws-cdk dev container finishes spinning up I could have a go -- imagine there'd be common code between the two we'd want to abstract into a shared place in the end